test/System/Vector2.cpp

```diff
@@ -37,6 +37,48 @@ TEST_CASE("sf::Vector2 class template - [system]")
+        {
+            CHECK(sf::Vector2f(0, sf::degrees(0)) == sf::Vector2f(0, 0));
+            CHECK(sf::Vector2f(0, sf::degrees(180)) == sf::Vector2f(0, 0));
+
+            CHECK(sf::Vector2f(1, sf::degrees(0)) == sf::Vector2f(1, 0));
+            CHECK(sf::Vector2f(1, sf::degrees(45)) == ApproxVec(std::sqrt(2.f) / 2.f, std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(1, sf::degrees(90)) == ApproxVec(0, 1));
+            CHECK(sf::Vector2f(1, sf::degrees(135)) == ApproxVec(-std::sqrt(2.f) / 2.f, std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(1, sf::degrees(180)) == ApproxVec(-1, 0));
+            CHECK(sf::Vector2f(1, sf::degrees(270)) == ApproxVec(0, -1));
+            CHECK(sf::Vector2f(1, sf::degrees(360)) == ApproxVec(1, 0));
+            CHECK(sf::Vector2f(1, sf::degrees(-90)) == ApproxVec(0, -1));
+            CHECK(sf::Vector2f(1, sf::degrees(-180)) == ApproxVec(-1, 0));
+            CHECK(sf::Vector2f(1, sf::degrees(-270)) == ApproxVec(0, 1));
+            CHECK(sf::Vector2f(1, sf::degrees(-360)) == ApproxVec(1, 0));
+
+            CHECK(sf::Vector2f(-1, sf::degrees(0)) == sf::Vector2f(-1, 0));
+            CHECK(sf::Vector2f(-1, sf::degrees(45)) == ApproxVec(-std::sqrt(2.f) / 2.f, -std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(-1, sf::degrees(90)) == ApproxVec(0, -1));
+            CHECK(sf::Vector2f(-1, sf::degrees(135)) == ApproxVec(std::sqrt(2.f) / 2.f, -std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(-1, sf::degrees(180)) == ApproxVec(1, 0));
+            CHECK(sf::Vector2f(-1, sf::degrees(270)) == ApproxVec(0, 1));
+            CHECK(sf::Vector2f(-1, sf::degrees(360)) == ApproxVec(-1, 0));
+            CHECK(sf::Vector2f(-1, sf::degrees(-90)) == ApproxVec(0, 1));
+            CHECK(sf::Vector2f(-1, sf::degrees(-180)) == ApproxVec(1, 0));
+            CHECK(sf::Vector2f(-1, sf::degrees(-270)) == ApproxVec(0, -1));
+            CHECK(sf::Vector2f(-1, sf::degrees(-360)) == ApproxVec(-1, 0));
+
+            CHECK(sf::Vector2f(3.14f, sf::degrees(0)) == sf::Vector2f(3.14f, 0));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(45)) == ApproxVec(3.14f * std::sqrt(2.f) / 2.f, 3.14f * std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(90)) == ApproxVec(0, 3.14f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(135)) == ApproxVec(-3.14f * std::sqrt(2.f) / 2.f, 3.14f * std::sqrt(2.f) / 2.f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(180)) == ApproxVec(-3.14f, 0));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(270)) == ApproxVec(0, -3.14f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(360)) == ApproxVec(3.14f, 0));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(-90)) == ApproxVec(0, -3.14f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(-180)) == ApproxVec(-3.14f, 0));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(-270)) == ApproxVec(0, 3.14f));
+            CHECK(sf::Vector2f(3.14f, sf::degrees(-360)) == ApproxVec(3.14f, 0));
```

Comment:
Is there a particular reason why you use "nearly Pi"? It doesn't seem like you make use of any of its properties, so I wonder if another arbitrary magic number might be less suggestive.

Reply:
Haha yeah that was a bad idea. I'll change that.